client/src/components/TimePreferencesModal.js

```diff
@@ -0,0 +1,351 @@
+        text: '11pm - 12am'
+    },
+]
+export default class TimePreferencesModal extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            day: '',
+            selectedTimes: this.props.timePreferences || [],
+            uncommittedTimesForCurrentDay: [],
+            submitting: false
+        }
+        this.handleDayChange = this.handleDayChange.bind(this)
+        this.getSelectedTimes = this.getSelectedTimes.bind(this)
+        this.removeSelectedTime = this.removeSelectedTime.bind(this)
+        this.handleTimeSlotChange = this.handleTimeSlotChange.bind(this)
+        this.commitAvailability = this.commitAvailability.bind(this)
+        this.submitPreferences = this.submitPreferences.bind(this)
+        this.hasNewPreferences = this.hasNewPreferences.bind(this)
+    }
+
+    hasNewPreferences() {
+        let oldTimeIds = this.props.timePreferences && this.props.timePreferences.map(time => time.id)
+        let newTimeIds = this.state.selectedTimes.map(time => time.id)
+        if (newTimeIds.length !== oldTimeIds.length) {
+            return true
+        }
+        return newTimeIds.filter(time => !oldTimeIds.includes(time)).length
+    }
+
+    submitPreferences(e) {
+        e.preventDefault()
+        this.setState({
+            submitting: true
+        }, async () => {
+            let payload = {
+                selectedTimes: this.state.selectedTimes,
+                email: this.props.email
+            }
+            try {
+                // TODO: Need to add end-point
+                const result = await makeCall(payload, '/alumni/updateTimePreferences', 'post')
+                if (!result || result.error) {
+                    this.setState({
+                        submitting: false
+                    }, () => {
+                        swal({
+                            title: "Error!",
+                            text: "There was an error updating your time preferences, please try again.",
+                            icon: "error",
+                        });
+                    })
+                } else {
+                    this.setState({
+                        submitting: false
+                    }, () => {
+                        swal({
+                            title: "Done!",
+                            text: "Your time preferences were successfully updated!",
+                            icon: "success",
+                        })
+                    })
+                    
+                }
+            } catch (e) {
+                this.setState({
+                    submitting: false
+                }, () => {
+                    console.log("Error: TimePreferencesModal#submitPreferences", e);
+                })
+            }
+        })
+    }
+
+    handleDayChange(e, {value}) {
+        e.preventDefault();
+        this.setState({
+            day: value
+        })
+    }
+
+    removeSelectedTime(e, timeId) {
```

Comment:
Triggered when `x` is clicked on a `pill` representing a new time selection (at top)